### DISCUSSION TITLE
examples/http3_client.py: How to manually control the data transmission rate.

### DISCUSSION THREAD

**Discussion by Anvilian** [aiortc/aioquic#527] [Q&A]
I would like to send data at specific time interval, and I am trying to change line 233 of examples/http3_client.py to this:
`for i in range(len(data_list)):
	self._http.send_data(
		stream_id=stream_id, data=data_list[i], end_stream=False
	)
	self.transmit()
	time.sleep(0.05)`
But it didn't work as expected. In the loop, it only assembles the datagram and does not actually send it out. And aioquic will send all the packets at once after the loop ends. How can I achieve sending data at specific time interval.

**Comment by jorektheglitch** [comment]
@Anvilian
> `time.sleep(0.05)`

Consider not to use syncronyous `time.wait` in async code. Try to replace tris line with `await asyncio.wait(0.05)`, maybe it suddenly gets work.

**Comment by rthalley** [comment]
Just a note that QUIC connections are also subject to flow control, so there's no guarantee that anything will be sent when you call transmit.
